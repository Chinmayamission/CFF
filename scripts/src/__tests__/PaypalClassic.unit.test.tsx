import React from "react";
import { shallow, mount, render } from 'enzyme';
import PaypalClassic from "src/form/confirmation/PaypalClassic";
import {Provider} from "react-redux";
import store from "src/store";

it('renders default payment form', () => {
  let props = {
    "onPaymentStarted": e => e,
    "paymentStarted": false,
    "paymentInfo_owed": {
      "currency": "USD",
      "items": [{
        "name": "Name",
        "description": "Description",
        "amount": 40,
        "quantity": 1
      }]
    },
    "paymentInfo_received": {
      "currency": "USD",
      "total": 0
    },
    "paymentInfo": {
      "currency": "USD",
      "items": [{
        "name": "Name",
        "description": "Description",
        "amount": 40,
        "quantity": 1
      }]
    },
    "paymentMethodInfo": {

    },
    "key": "k",    // must be unique.
    "onPaymentComplete": e => e,
    "onPaymentError": e => e,
    "responseId": "r123456",
    "formId": "f123456",
    "formData": {

    },
    "paymentMethodName": "paypal_classic"
    // todo: get user's entered data.
  };
  const wrapper = render(
    <Provider store={store}><PaypalClassic {...props} /></Provider>
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders subscription payment form', () => {
  let props = {
    "onPaymentStarted": e => e,
    "paymentStarted": false,
    "paymentInfo_owed": {
      "currency": "USD",
      "items": [{
        "name": "Name",
        "description": "Description",
        "amount": 40,
        "quantity": 1,
        "recurrence": "5Y"
      }]
    },
    "paymentInfo_received": {
      "currency": "USD",
      "total": 0
    },
    "paymentInfo": {
      "currency": "USD",
      "items": [{
        "name": "Name",
        "description": "Description",
        "amount": 40,
        "quantity": 1
      }]
    },
    "paymentMethodInfo": {

    },
    "key": "k",    // must be unique.
    "onPaymentComplete": e => e,
    "onPaymentError": e => e,
    "responseId": "r123456",
    "formId": "f123456",
    "formData": {

    },
    "paymentMethodName": "paypal_classic"
    // todo: get user's entered data.
  };
  const wrapper = render(
    <Provider store={store}><PaypalClassic {...props} /></Provider>
  );
  expect(wrapper).toMatchSnapshot();
});