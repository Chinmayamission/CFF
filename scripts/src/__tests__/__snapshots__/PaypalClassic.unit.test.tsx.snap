// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default payment form 1`] = `
<form
  action="https://www.paypal.com/cgi-bin/webscr"
  method="post"
>
  <input
    name="cmd"
    type="hidden"
    value="_cart"
  />
  <input
    name="custom"
    type="hidden"
    value="r123456"
  />
  <input
    name="charset"
    type="hidden"
    value="utf-8"
  />
  <input
    name="upload"
    type="hidden"
    value="1"
  />
  <input
    name="business"
    type="hidden"
  />
  <input
    name="currency_code"
    type="hidden"
    value="USD"
  />
  <input
    name="notify_url"
    type="hidden"
    value="http://localhost:8001/responses/r123456/ipn"
  />
  <input
    name="return"
    type="hidden"
    value="http://localhost/"
  />
  <input
    name="cancel_return"
    type="hidden"
    value="http://localhost/"
  />
  <div>
    <input
      name="item_name_1"
      type="hidden"
      value="Name"
    />
    <input
      name="item_number_1"
      type="hidden"
      value="Description"
    />
    <input
      name="quantity_1"
      type="hidden"
      value="1"
    />
    <input
      name="amount_1"
      type="hidden"
      value="40"
    />
  </div>
  <input
    name="discount_amount_cart"
    type="hidden"
    value="0"
  />
  <input
    name="image_url"
    type="hidden"
  />
  <input
    name="first_name"
    type="hidden"
  />
  <input
    name="last_name"
    type="hidden"
  />
  <input
    name="address1"
    type="hidden"
  />
  <input
    name="address2"
    type="hidden"
  />
  <input
    name="city"
    type="hidden"
  />
  <input
    name="state"
    type="hidden"
  />
  <input
    name="zip"
    type="hidden"
  />
  <input
    name="night_phone_a"
    type="hidden"
  />
  <input
    name="night_phone_b"
    type="hidden"
  />
  <input
    name="night_phone_c"
    type="hidden"
  />
  <input
    name="email"
    type="hidden"
  />
  <input
    class="btn btn-primary"
    type="submit"
    value="Pay Now with Paypal"
  />
</form>
`;

exports[`renders subscription payment form 1`] = `
<form
  action="https://www.paypal.com/cgi-bin/webscr"
  method="post"
>
  <input
    name="cmd"
    type="hidden"
    value="_xclick-subscriptions"
  />
  <input
    name="custom"
    type="hidden"
    value="r123456"
  />
  <input
    name="charset"
    type="hidden"
    value="utf-8"
  />
  <input
    name="upload"
    type="hidden"
    value="1"
  />
  <input
    name="business"
    type="hidden"
  />
  <input
    name="currency_code"
    type="hidden"
    value="USD"
  />
  <input
    name="notify_url"
    type="hidden"
    value="http://localhost:8001/responses/r123456/ipn"
  />
  <input
    name="return"
    type="hidden"
    value="http://localhost/"
  />
  <input
    name="cancel_return"
    type="hidden"
    value="http://localhost/"
  />
  <input
    name="image_url"
    type="hidden"
  />
  <input
    name="first_name"
    type="hidden"
  />
  <input
    name="last_name"
    type="hidden"
  />
  <input
    name="address1"
    type="hidden"
  />
  <input
    name="address2"
    type="hidden"
  />
  <input
    name="city"
    type="hidden"
  />
  <input
    name="state"
    type="hidden"
  />
  <input
    name="zip"
    type="hidden"
  />
  <input
    name="night_phone_a"
    type="hidden"
  />
  <input
    name="night_phone_b"
    type="hidden"
  />
  <input
    name="night_phone_c"
    type="hidden"
  />
  <input
    name="email"
    type="hidden"
  />
  <input
    class="btn btn-primary"
    type="submit"
    value="Pay Now with Paypal"
  />
  <div>
    <input
      name="item_name"
      type="hidden"
      value="Name"
    />
    <input
      name="a3"
      type="hidden"
      value="40"
    />
    <input
      name="p3"
      type="hidden"
      value="1"
    />
    <input
      name="t3"
      type="hidden"
      value="M"
    />
    <input
      name="no_note"
      type="hidden"
      value="1"
    />
    <input
      name="src"
      type="hidden"
      value="1"
    />
  </div>
</form>
`;

exports[`renders subscription payment form with max number of recurrenceTimes 1`] = `
<form
  action="https://www.paypal.com/cgi-bin/webscr"
  method="post"
>
  <input
    name="cmd"
    type="hidden"
    value="_xclick-subscriptions"
  />
  <input
    name="custom"
    type="hidden"
    value="r123456"
  />
  <input
    name="charset"
    type="hidden"
    value="utf-8"
  />
  <input
    name="upload"
    type="hidden"
    value="1"
  />
  <input
    name="business"
    type="hidden"
  />
  <input
    name="currency_code"
    type="hidden"
    value="USD"
  />
  <input
    name="notify_url"
    type="hidden"
    value="http://localhost:8001/responses/r123456/ipn"
  />
  <input
    name="return"
    type="hidden"
    value="http://localhost/"
  />
  <input
    name="cancel_return"
    type="hidden"
    value="http://localhost/"
  />
  <input
    name="image_url"
    type="hidden"
  />
  <input
    name="first_name"
    type="hidden"
  />
  <input
    name="last_name"
    type="hidden"
  />
  <input
    name="address1"
    type="hidden"
  />
  <input
    name="address2"
    type="hidden"
  />
  <input
    name="city"
    type="hidden"
  />
  <input
    name="state"
    type="hidden"
  />
  <input
    name="zip"
    type="hidden"
  />
  <input
    name="night_phone_a"
    type="hidden"
  />
  <input
    name="night_phone_b"
    type="hidden"
  />
  <input
    name="night_phone_c"
    type="hidden"
  />
  <input
    name="email"
    type="hidden"
  />
  <input
    class="btn btn-primary"
    type="submit"
    value="Pay Now with Paypal"
  />
  <div>
    <input
      name="item_name"
      type="hidden"
      value="Name"
    />
    <input
      name="a3"
      type="hidden"
      value="40"
    />
    <input
      name="p3"
      type="hidden"
      value="1"
    />
    <input
      name="t3"
      type="hidden"
      value="M"
    />
    <input
      name="no_note"
      type="hidden"
      value="1"
    />
    <input
      name="src"
      type="hidden"
      value="1"
    />
    <input
      name="srt"
      type="hidden"
      value="12"
    />
  </div>
</form>
`;
